uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Form/GeneralSettings.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+require_once 'CRM/Core/Form.php';
+
+/**
+ * Form controller class
+ *
+ * @see http://wiki.civicrm.org/confluence/display/CRMDOC43/QuickForm+Reference
+ */
+class CRM_HRLeaveAndAbsences_Form_GeneralSettings extends CRM_Core_Form {
+
+  /**
+   * @var array
+   *   The filter to pass to the setting.getfields API
+   */
+  private $_settingFilter = ['group' => 'leaveandabsence'];
+
+  /**
+   * @var array
+   *   An array to store settings once it has been retrieved from the settings API
+   */
+  private $_settings = [];
+
+  private $_submittedValues = [];
+
+  function buildQuickForm() {
+    $settings = $this->getFormSettings();
+
+    foreach ($settings as $name => $setting) {
+      if ($name == 'relationship_allowed_to_approve_leave') {
+        $this->$setting['html_type'](
+          $name,
+          [
+            'options' => $this->getRelationshipTypes(),
+            'multiple' => true,
+            'label' => $setting['label'],
+            'style' => $setting['html_attributes']['style'],
+          ],
+          true
+        );
+      }
+    }
+
+    $this->addButtons($this->getAvailableButtons());
+
+    // export form elements to template
```

Comment:
no need for this comment. The code is already self explanatory (for someone who knows how forms and pages handle templates :D )